pkg/recipes/terraform/config/providers/types.go

```diff
@@ -34,10 +34,10 @@ type Provider interface {
 	BuildConfig(ctx context.Context, envConfig *recipes.Configuration) (map[string]any, error)
 }
 
-// GetSupportedTerraformProviders returns a map of Terraform provider names to provider config builder.
-// Providers represent Terraform providers for which Radius generates custom provider configurations.
+// GetUCPSupportedTerraformProviders returns a map of Terraform provider names with configuration details stored in UCP, to provider config builder.
+// These providers represent Terraform providers for which Radius generates custom provider configurations based on credentials stored with UCP.
```

Comment:
It also adds non credential provider configuration (subscription id for azure, for example)

```suggestion
// These providers represent Terraform providers for which Radius generates custom provider configurations based on credentials stored with UCP and providers configured on the Radius environment.
```